homeassistant/components/feedreader/coordinator.py

```diff
@@ -0,0 +1,247 @@
+"""Data update coordinator for RSS/Atom feeds."""
+
+from __future__ import annotations
+
+from calendar import timegm
+from datetime import datetime, timedelta
+from logging import getLogger
+import os
+import pickle
+from time import gmtime, struct_time
+
+import feedparser
+
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.storage import Store
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+from homeassistant.util import dt as dt_util
+
+from .const import DELAY_SAVE, DOMAIN, EVENT_FEEDREADER, STORAGE_VERSION
+
+_LOGGER = getLogger(__name__)
+
+
+class FeedReaderCoordinator(DataUpdateCoordinator):
+    """Abstraction over Feedparser module."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        url: str,
+        scan_interval: timedelta,
+        max_entries: int,
+        storage: StoredData,
+    ) -> None:
+        """Initialize the FeedManager object, poll as per scan interval."""
+        super().__init__(
+            hass=hass,
+            logger=_LOGGER,
+            name=f"{DOMAIN} {url}",
+            update_interval=scan_interval,
+        )
+        self._url = url
+        self._max_entries = max_entries
+        self._feed: feedparser.FeedParserDict | None = None
+        self._firstrun = True
```

Comment:
you don't need to track this, you can derive this from `self._last_entry_timestamp is None`

Reply:
4fd68511fce5b6c0fe1ee16443368b56e0a803c2